lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerFactory.java

```diff
@@ -51,32 +56,39 @@ private DefaultRemoteManagerFactory(
       @Parameter(RemoteConfiguration.NumberOfTries.class) final int numberOfTries,
       @Parameter(RemoteConfiguration.RetryTimeout.class) final int retryTimeout,
       final LocalAddressProvider localAddressProvider,
-      final TransportFactory tpFactory) {
+      final TransportFactory tpFactory,
+      final TcpPortProvider tcpPortProvider,
+      final Injector injector) {
+    this.injector = injector;
     this.codec = codec;
     this.errorHandler = errorHandler;
     this.orderingGuarantee = orderingGuarantee;
     this.numberOfTries = numberOfTries;
     this.retryTimeout = retryTimeout;
     this.localAddressProvider = localAddressProvider;
-    this.tpFactory = tpFactory;
+    this.transportFactory = tpFactory;
+    this.tcpPortProvider = tcpPortProvider;
   }
 
-  // TODO[REEF-547]: This method uses deprecated DefaultRemoteManagerImplementation constructor.
   @Override
   public RemoteManager getInstance(final String name) {
-    return new DefaultRemoteManagerImplementation(name,
-        DefaultRemoteManagerImplementation.UNKNOWN_HOST_NAME, // Indicate to use the localAddressProvider
-        0, // Indicate to use the tcpPortProvider
-        this.codec,
-        this.errorHandler,
-        this.orderingGuarantee,
-        this.numberOfTries,
-        this.retryTimeout,
-        this.localAddressProvider,
-        this.tpFactory);
+    try {
+      final Injector newInjector = injector.forkInjector();
+      newInjector.bindVolatileParameter(RemoteConfiguration.ManagerName.class, name);
```

Comment:
Have you considered creating a `Configuration` with the bindings and passing that into the `forkInjector()` call instead of using `bindVolatile`? That is somewhat safer.

Also, bindings to things injected into this `DefaultRemoteManagerFactory` should already be in the forked injector. I don't think you need to `bindVolatile` them. In fact, I am pretty sure those calls should throw exceptions.